### ISSUE TITLE
Implement PullRequest MergeableState field

### ISSUE THREAD

**Issue by ryangribble** [octokit/octokit.net#1763]
This is currently an undocumented field (likely to be added to docs in the future) but provides very useful information about whether a PR can be merged or not, and what the reasons preventing merge actually are.

This field should be implemented as a new `Enum` class, using our standard `Parameter()` property to map to the api string values.  In the `PullRequest` class it will be wrapped by our `StringEnum<T>` wrapper to guard against the addition of more values upstream.  It most likely will need to be declared nullable, as it probably isn't returned in `GetAll` responses, that only return a subset of `PullRequest` fields

The following values are possible

`dirty`: Merge conflict. Merging will be blocked

`unknown`: Mergeability was not checked yet. Merging will be blocked

`blocked`: Blocked by a failing/missing required status check.

`behind`: Head branch is behind the base branch. Only if required status checks is enabled but loose policy is not. Merging will be blocked.

`unstable`: Failing/pending commit status that is not part of the required status checks. Merging is allowed (yellow box).

`has_hooks`: GitHub Enterprise only, if a repo has custom pre-receive hooks. Merging is allowed (green box).

`clean`: No conflicts, everything good. Merging is allowed (green box).

Some references:
https://github.com/octokit/octokit.rb/issues/626
https://github.com/google/go-github/pull/820#issuecomment-354459645
https://platform.github.community/t/documentation-about-mergeable-state/4259
